src/core/main/worker/worker_main.ts

```diff
@@ -73,6 +74,7 @@ export default function initializeWorkerMain() {
   // TODO allow worker-side feature-switching? Not sure how
   const dashWasmParser = new DashWasmParser();
   features.dashParsers.wasm = dashWasmParser;
+  features.dashParsers.fastJs = DashFastJsParser;
   features.transports.dash = createDashPipelines;
 
```

Comment:
Is there a reason to prefer assigning those property over calling the exposed methods in `src/features/list` like `addDashFeature`, `dashWasmFeature` .. ?

Reply:
That's a good question and I've thought about it.

However it seems that features exported in `src/features/list`, because they are the exact same files imported by applications, assume a main thread environment. This is to reduce the number of features to import for an application (the end goal being a simpler API).

As such, the importing `src/features/lib/dash.ts` also adds a `MediaSourceContentInitializer` to play content in main thread and a `MainCodecSupportProber` to perform codec checks on main thread.

Because we don't want those in a worker, in the end I kept just manually selecting the exact feature we want in `worker_main.ts`